test/test_repository_specification.py

```diff
@@ -0,0 +1,22 @@
+import os
+
+from rosdistro.repository_specification import RepositorySpecification
+
+def test_repository_specification():
+    data = { 'type': 'git', 'url': 'https://github.com/ros/catkin.git' }
+    r = RepositorySpecification("test", data)
+    assert r.get_data() == data
+    assert r.version == None
+    assert r.get_url_parts() == ('github.com', 'ros/catkin')
+
+    r.url = 'http://github.com/ros/catkin'
+    assert r.get_url_parts() == ('github.com', 'ros/catkin')
```

Comment:
I would have made this only accept the `.git` suffix, but there are URLs in `ros/rosdistro` that haven't got it.